### PR TITLE
Add new Midnight primers

### LINKED ISSUE #14: Switch to a BED file for primers/amplicons
Currently, we are using two separate files to indicate our primers:

- `reference/primers.fasta`: This is used by BBDuk in the `TRIM_PRIMERS` step. We can generate this using the [`getfasta` command from the bedtools suite](https://bedtools.readthedocs.io/en/latest/content/tools/getfasta.html) and the FASTA provided by Nextclade.
- `reference/midnight_primers_nextclade.tsv`: This is [used by Nextclade](https://docs.nextstrain.org/projects/nextclade/en/stable/user/input-files.html#pcr-primers). I would have to write a script that can combine the BED file, the FASTA and GFF (v2) files provided by Nextclade, and other user-provided information such as "institution" (for some reason).

### LINKED ISSUE #20: Add new primers to Midnight set
These are primers to be used **in addition to** our current suite (it's helpful to think of these collectively as a "Midnight v1.2 primer set"). This will throw a slight wrench in my plans with #14, as there's no standardized way to specify a sequence in a BED file. (Perhaps we could take a look at using a modified ARTIC-style TSV file, with every possible data point we may need, as a single source of truth.)

# New Primers
The following primer is part of the "Midnight v1.1" primer set:
- \>SARSCoV_1200_28_LEFT_OMICRON (C27807T)
  TTTGTGCTTTTTAGCCTTTCTG**T**T

In addition to the previous primer, the following three primers are part of the "Midnight v1.2" primer set:
- \>SARSCoV_1200_22_RIGHT_OMICRON (G22599A)
  AACAGATGCAAAT**T**TGGTGGCG
- \>SARSCoV_1200_23_LEFT_OMICRON (C22522T)
  ACTTTAGAGT**T**CAACCAACAGAATCT
- \>SARSCoV_1200_26_LEFT_OMICRON (C25708T)
  GCCTTGAAGCCCCTTTT**T**TCTA

# References

- https://twitter.com/freed_nikki/status/1464477513107730433?s=20
- https://twitter.com/DrT1973/status/1483924384238628866?s=20

### PR DESCRIPTION
**Pull request by fanninpm**
Currently, the primers are from v1.0, but that is subject to change following a review from @friesac.

There are also plans to output a primer BED file and an amplicon BED file, but they haven't been implemented yet.

Fixes #20. Closes #14.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -12,15 +12,15 @@ if (!params.skip_performance_excel) {
 }
 
 params.ivar_gff_gzip_file = "https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/009/858/895/GCA_009858895.3_ASM985889v3/GCA_009858895.3_ASM985889v3_genomic.gff.gz"
-params.primer_fasta = workflow.projectDir + "/reference/primers.fasta"
+params.primer_tsv = workflow.projectDir + "/reference/Midnight_Primers_v1.0.tsv"
 params.adapter_fasta = workflow.projectDir + "/reference/adapters.fa.gz"
 params.nextclade_primers = workflow.projectDir + "/reference/midnight_primers_nextclade.csv"
```
Comment (fanninpm) [inline main.nf:17]: TODO: get rid of this (or make this optional)

Comment (friesac) [reply main.nf:17]: If the question is related to the new primers moving from v1 to v2 then we are set there. However, I think you've been running with v2 lately through mad river if I'm not mistaken.

**Review by friesac — APPROVED** [review]
Seems good to me. Since we could envision lots of different amplicon approaches in the coming months it might be good to have a test process designed to make sure we can manually check the fasta to bed file or vice versa. I suppose it would just become standard practice to run the `process_primer_reference.py` prior to initiating any workflow?

File: reference/IDT_Midnight_Primers_v2.0.tsv [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+13	LEFT	1	ACCTCTTACAACAGCAGCCAAAC	+	12450	12473
+13	RIGHT	1	CGTCCTTTTCTTGGAAGCGACA	-	13621	13599
+15	LEFT	1	TTTTAAGGAATTACTTGTGTATGCTGCT	+	14540	14568
+15	RIGHT	1	ACACACAACAGCATCGTCAGAG	-	15735	15713
+17	LEFT	1	TCAAGCTTTTTGCAGCAGAAACG	+	16624	16647
+17	RIGHT	1	CCAAGCAGGGTTACGTGTAAGG	-	17754	17732
+19	LEFT	1	GGCACATGGCTTTGAGTTGACA	+	18596	18618
+19	RIGHT	1	CCTGTTGTCCATCAAAGTGTCCC	-	19678	19655
+21	LEFT	1	TCTGTAGTTTCTAAGGTTGTCAAAGTGA	+	20553	20581
+21	RIGHT	1	GCAGGGGGTAATTGAGTTCTGG	-	21642	21620
```
Comment (friesac) [inline reference/IDT_Midnight_Primers_v2.0.tsv:23]: Are you considering the ARTIC 71R inclusion in the IDT Midnight V2? Is that what you are referring to with 21 RIGHT_OMICRON?

Comment (fanninpm) [reply reference/IDT_Midnight_Primers_v2.0.tsv:23]: I'm going to rename the primer to (hopefully) alleviate confusion.

**Pull request merged**